src/main/io/osd.c

```diff
@@ -484,6 +485,28 @@ static uint16_t osdConvertRSSI(void)
     return constrain(getRSSI() * 100 / RSSI_MAX_VALUE, 0, 99);
 }
 
+/**
+* Converts temperature into a string based on the current unit system
+* postfixed with a symbol to indicate the unit used.
+* @param temperature Raw temperature (i.e. as taken from getCurrentTemperature() in degC)
+*/
+static void osdFormatTemperatureSymbol(char *buff, float temperature)
+{
+    int units_symbol;
+    switch ((osd_unit_e)osdConfig()->units) {
+        case OSD_UNIT_IMPERIAL:
+            units_symbol = SYM_TEMP_F;
+            temperature = (temperature * (9/5)) + 32;
+            break;
+        case OSD_UNIT_UK:
+            FALLTHROUGH;
+        case OSD_UNIT_METRIC:
+            units_symbol = SYM_TEMP_C;
+            break;
+    }
+    tfp_sprintf(buff, "%d%c", (int) temperature, units_symbol);
```

Comment:
Using a bare `%d` will leave ghosting characters at the back when the formatted value shortens. Please, use `osdFormatCentiNumber()`, it will take care of formatting in fixed width for you. Use 3 characters, so we can format from -99 to 999 and allow up to 2 decimals to fill the unused space.

Reply:
I'm having a bit of an hard time understanding its use with a non-decimal number - as we agreed before, I think decimals will just use up precious screen space in this case. Would this be a viable solution? I can't test it in this very moment.

```c
osdFormatCentiNumber(buff, (int) temperature, 0, 0, 0, 3);
buff[3] = units_symbol;
buff[4] = '\0';
```

Reply:
Thing is, in order to have constant width you need to either add spaces or decimals, so it will always take the space of the “longest” value you want to represent. IMHO, we want to represent up to 3 digits to be able to write negative temperatures lower than -9. Do you think we should contaron the range to [-99, 999] or would [-9, 99] suffice?

Reply:
[-9, 99] won't be sufficient, 99°F = 37.2°C, and during testing my Matek F405-Wing happily went over 60°C. I'd say we should go with [-99, 999], with no decimals on screen at any time. If padding needs to be added, spaces won't clutter the OSD as decimals would.